tracer/src/Datadog.Tracer.Native/tracer_tokens.cpp

```diff
@@ -157,6 +157,8 @@ const shared::WSTRING& TracerTokens::GetCallTargetRefStructType()
 
 HRESULT TracerTokens::EnsureBaseCalltargetTokens()
 {
+    std::lock_guard<std::recursive_mutex> guard(metadata_mutex);
```

Comment:
Nice! Maybe you can add the same guard in `debugger_tokens.cpp` & `fault_tolerant_tokens.cpp`?

Reply:
Good call. Done in https://github.com/DataDog/dd-trace-dotnet/pull/5836/commits/5fb4e1d59bea320529107137fa783b0622484089